src/status_im/chat/handlers/faucet.cljs

```diff
@@ -0,0 +1,60 @@
+(ns status-im.chat.handlers.faucet
+  (:require [re-frame.core :refer [dispatch]]
+            [status-im.utils.handlers :refer [register-handler] :as u]
+            [status-im.utils.utils :refer [http-get]]
+            [status-im.utils.random :as random]
+            [status-im.constants :refer [console-chat-id
+                                         text-content-type]]
+            [status-im.i18n :refer [label]]
+            [taoensso.timbre :as log]
+            [goog.string :as gstring]
+            goog.string.format))
+
+(def faucets
+  [{:name     "http://faucet.ropsten.be:3001"
+    :type     :api
+    :api-url  "http://faucet.ropsten.be:3001/donate/0x%s"}
+   {:name       "https://zerogox.com/ethereum/wei_faucet"
+    :type       :prefill
+    :prefill-js "document.getElementById('edit-ethereum-address').value='%s';"}])
+
+(defn faucet-by-name [faucet-name]
+  (->> faucets
+       (filter #(= (:name %) faucet-name))
+       (first)))
+
+(defn received-message [content]
+  (dispatch [:received-message
+             {:message-id   (random/id)
+              :content      content
+              :content-type text-content-type
+              :outgoing     false
+              :chat-id      console-chat-id
+              :from         console-chat-id
+              :to           "me"}]))
+
+(defmulti open-faucet (fn [_ _ {:keys [type]}] type))
+
+(defmethod open-faucet :api
+  [faucet-name current-address {:keys [api-url]}]
+  (let [api-url (gstring/format api-url current-address)]
+    (http-get api-url
+              #(received-message (label :t/faucet-success))
+              #(received-message (label :t/faucet-error)))))
+
+(defmethod open-faucet :prefill
+  [faucet-name current-address {:keys [prefill-js]}]
+  (let [prefill-js  (gstring/format prefill-js current-address)
+        web-view-js (gstring/format "window.onload = function(){ %s };" prefill-js)]
```

Comment:
Why do we need to make this magic, is it possible just to send request using something like this https://github.com/status-im/status-react/blob/3122cb7ac3c63ac500d9a6147ce0afa9a623b8c5/src/status_im/utils/utils.cljs#L21-L21 ? Maybe, in this case, we will be able to get some useful info from response